### PR TITLE
Add basic Logging and Config reading

### LINKED ISSUE #2: Add Logging
Log down visiting IP addresses and incoming requests as well as errors into info_log and error_log

### PR DESCRIPTION
**Pull request by LLeon360**
Added a config to designate where logs go and add logging to log some information and timestamps for successful requests as well as errors.

### CODE REVIEW

**Comment by LLeon360** [comment]
Currently the dev task for `deno task dev` just uses --allow-write=. I'm not sure what the best way is to put the log directory from config into here or if I should just not use a config.

**Comment by epsteinmark** [comment]
> Currently the dev task for `deno task dev` just uses --allow-write=. I'm not sure what the best way is to put the log directory from config into here or if I should just not use a config.

It'd probably be smart to use a config, you can use `--allow-write=./log/` if you'd like to specify a directory

**Comment by LLeon360** [comment]
I have a config file called 
config.json that looks like:
```json
{
    "logging" : {
        "logDir" : "logs/",
        "infoLog" : "info_log.log",
        "errorLog" : "error_log.log"
    }
}
```
just not sure what the best way is to get logDir and put it into the allow-write.

**Comment by epsteinmark** [comment]
> I have a config file called config.json that looks like:
> just not sure what the best way is to get logDir and put it into the allow-write.



JSON does not have a way to fetch values from other JSON files without writing external code (it would be quite the security vulnerability otherwise!). It also can't self reference, so you'll have to manually write `--allow-write=./logs/`

That being said, deno.json doesn't seem to be upset if you add your own fields that are not their own, so you can just merge your config.json into deno.json (although this is a personal decision of whether to have a change point in multiple files or just one vs separating deno and your normal config)

File: logging.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    };
+
+    // if the logs directory does not exist, create it
+    try {
+        await Deno.mkdir(config.logging.logDir);
+    } catch (error) {
+        if (!(error instanceof Deno.errors.AlreadyExists)) {
+            console.error(`Failed to create log directory: ${error}`);
+        }
+    }
```
Comment (bkrl) [inline logging.ts:47]: You can just use the [`recursive` option](https://deno.land/api@v1.43.3?s=Deno.MkdirOptions#prop_recursive).

Comment (LLeon360) [reply logging.ts:47]: Wait true

**Pull request merged**